src/sagas/RootSaga.js

```diff
@@ -66,35 +69,60 @@ export function* handleFetchProfile(action) {
 }
 
 export function* handleSaveProfile(action) {
-  const { username, accountDrafts, visibilityDrafts } = yield select(handleSaveProfileSelector);
-
   try {
+    const { username, drafts, preferences } = yield select(handleSaveProfileSelector);
+
+    const accountDrafts = keepKeys(drafts, [
+      'bio',
+      'courseCertificates',
+      'country',
+      'education',
+      'languageProficiencies',
+      'name',
+      'socialLinks',
+    ]);
+
+    const preferencesDrafts = keepKeys(drafts, [
+      'visibilityBio',
+      'visibilityCourseCertificates',
+      'visibilityCountry',
+      'visibilityEducation',
+      'visibilityLanguageProficiencies',
+      'visibilityName',
+      'visibilitySocialLinks',
+    ]);
+
+    if (Object.keys(preferencesDrafts).length > 0) {
+      preferencesDrafts.accountPrivacy = 'custom';
+    }
+
     yield put(saveProfileBegin());
     let accountResult = null;
     // Build the visibility drafts into a structure the API expects.
-    const preferences = {
-      visibility: visibilityDrafts,
-    };
 
     if (Object.keys(accountDrafts).length > 0) {
       accountResult = yield call(ProfileApiService.patchProfile, username, accountDrafts);
     }
 
-    if (Object.keys(visibilityDrafts).length > 0) {
-      yield call(ProfileApiService.patchPreferences, username, preferences);
+    let preferencesResult = preferences; // assume it hasn't changed.
```

Comment:
If we don't make the prefs call (because there were no changes), just assume the "result" is what our preferences currently are.